Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm

```diff
@@ -325,11 +326,26 @@ void requestLinkDecorationFilteringData(LinkFilteringRulesCallback&& callback)
     return map;
 }
 
+void StorageAccessPromptQuirkController::initialize()
+{
+    if (m_wasInitialized)
+        return;
+
+    updateQuirks([this] {
+        m_observers.forEach([](auto& observer) {
+            observer.invokeCallback();
+        });
+    });
+    m_wasInitialized = true;
+}
+
 void StorageAccessPromptQuirkController::updateQuirks(CompletionHandler<void()>&& completionHandler)
 {
     ASSERT(RunLoop::isMain());
     if (!WebKit::canUseWebPrivacyFramework() || ![PAL::getWPResourcesClass() instancesRespondToSelector:@selector(requestStorageAccessPromptQuirksData:completionHandler:)]) {
-        completionHandler();
+        RunLoop::main().dispatch([completionHandler = WTFMove(completionHandler)]() mutable {
+            completionHandler();
+        });
```

Comment:
Nit - does it work if we just call it like this? (Might be a tiny bit cleaner)
```suggestion
        RunLoop::main().dispatch(WTFMove(completionHandler));
```

Reply:
Smart :)